### PR TITLE
[WIP]Add user-request to audit policy and profile

### PR DESCRIPTION
**Pull request by EmilyM1**
This PR adds the userrequest profile and policy and associated unit tests. This policy adds the ability to filter audit logs on the resource requests of system:authenticated:oauth users.

### CODE REVIEW

File: pkg/operator/apiserver/audit/bindata/bindata.go [hunk 1]
```diff
@@ -311,7 +311,29 @@ data:
...
+    - level: RequestResponse
+      resources:
+      - group: "oauth.openshift.io"
+        resources: ["oauthaccesstokens"]
+      verbs:
+      - create
+      - delete
+    - level: Metadata
+      userGroups:
+      - "system:authenticated:oauth"s
```
Comment (tkashem) [inline pkg/operator/apiserver/audit/bindata/bindata.go:333]: typo?

Comment (EmilyM1) [reply pkg/operator/apiserver/audit/bindata/bindata.go]: Yes. It also needed a type change.

**Comment by EmilyM1** [comment]
@tkashem can you please review this? Thank You!

**Comment by tkashem** [comment]
@EmilyM1 the yaml looks good.

A couple of suggestions:
- please squash the commits
- the unit tests are failing, please fix it - https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_library-go/982/pull-ci-openshift-library-go-master-unit/1362131773118484480

**Comment by tkashem** [comment]
looks good, I have a couple of minor suggestions:
- add a description of the PR, why we need it
- reformat the commit: the subject has bled into the body.
-

File: pkg/operator/apiserver/audit/manifests/audit-policies-cm.yaml [hunk 2]
```diff
@@ -255,4 +255,50 @@ data:
...
+      name: UserRequests
+    # Don't generate audit events for all requests in RequestReceived stage.
+    omitStages:
+    - "RequestReceived"
+    rules:
+    # Logs request and response for oauthaccess tokens.
+    - level: RequestResponse
+      resources:
+      - group: "oauth.openshift.io"
+        resources: ["oauthaccesstokens"]
```
Comment (deads2k) [inline pkg/operator/apiserver/audit/manifests/audit-policies-cm.yaml:273]: If I'm reading correctly, this seems to run counter to our goals of avoiding sensitive information in audit logs.  I think RequestResponse is all metadata plus the request and response bodies.  If you do this for oauthaccesstokens, I think the secrets are exposed in the log file.

**Comment by deads2k** [comment]
/hold

Let's get a unit test created that ensures we don't accidentally create a new policy that exposes information we're trying to avoid exposing.  oauthaccesstoken names, secret content, tokenreviews content (new).  The unit test should land in advance of this PR if my comment https://github.com/openshift/library-go/pull/982#discussion_r580536628 interprets the change correctly.

Also, is there an update to https://github.com/openshift/enhancements/blob/master/enhancements/kube-apiserver/audit-policy.md#proposal or somewhere describing the goals of our new policy.

File: pkg/operator/apiserver/audit/manifests/audit-policies-cm.yaml (hunk 2 above)
Comment (EmilyM1) [reply pkg/operator/apiserver/audit/manifests/audit-policies-cm.yaml:273]: I had grabbed this audit policy from the Jira ticket.
What change do you rec to fully suppress this?

File: pkg/operator/apiserver/audit/bindata/bindata.go [hunk 3]
```diff
@@ -311,7 +311,54 @@ data:
...
+      name: UserRequests
+    # Don't generate audit events for all requests in RequestReceived stage.
+    omitStages:
+    - "RequestReceived"
+    rules:
+    # Logs request and response for oauthaccess tokens.
+    - level: RequestResponse
+      resources:
+      - group: "oauth.openshift.io"
+        resources: ["oauthaccesstokens"]
```
Comment (sttts) [inline pkg/operator/apiserver/audit/bindata/bindata.go:329]: we cannot log these if there are still old tokens in the system. @stlaz is it time to finally delete old tokens?

Comment (stlaz) [reply pkg/operator/apiserver/audit/bindata/bindata.go:329]: I suppose it'd be the right time to do that

Comment (EmilyM1) [reply pkg/operator/apiserver/audit/bindata/bindata.go:329]: Should I go to -level: None here? I think that would be better.

**Pull request closed without merging**